### PR TITLE
Allow reading non-ORC tables when pushdown filter is enabled 

### PR DESCRIPTION
**Pull request by bhhari**
capture whether pushdown is enabled at query plan time and pass it into tableLayout to use it during execution

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -1784,7 +1784,8 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
                                 hivePartitionResult.getEnforcedConstraint(),
                                 hivePartitionResult.getBucketHandle(),
                                 hivePartitionResult.getBucketFilter(),
-                                createTableLayoutString(session, tableName, hivePartitionResult.getBucketHandle(), decomposedFilter.getRemainingExpression(), domainPredicate))),
+                                createTableLayoutString(session, tableName, hivePartitionResult.getBucketHandle(), decomposedFilter.getRemainingExpression(), domainPredicate),
+                                isPushdownFilterSupported(session, tableHandle))),
```
Comment (mbasmanova) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: this should always be true (it is inside the `pushdownFilter` method)

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 2]
```diff
@@ -1876,8 +1877,8 @@ public Void visitVariableReference(VariableReferenceExpression variable, Immutab
                                 hivePartitionResult.getEnforcedConstraint(),
                                 hiveBucketHandle,
                                 hivePartitionResult.getBucketFilter(),
-                                createTableLayoutString(session, handle.getSchemaTableName(), hivePartitionResult.getBucketHandle(), TRUE_CONSTANT, domainPredicate))),
-                hivePartitionResult.getUnenforcedConstraint()));
+                                createTableLayoutString(session, handle.getSchemaTableName(), hivePartitionResult.getBucketHandle(), TRUE_CONSTANT, domainPredicate),
+                                isPushdownFilterSupported(session, tableHandle))), hivePartitionResult.getUnenforcedConstraint()));
```
Comment (mbasmanova) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: - this should always be false 
- `hivePartitionResult.getUnenforcedConstraint()` needs to stay on a separate line

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java [hunk 3]
```diff
@@ -654,6 +654,16 @@ public void testPathColumn()
         assertQuerySucceeds(session, "SELECT linenumber, \"$path\" FROM lineitem WHERE length(\"$path\") % 2 = linenumber % 2");
     }
 
+    @Test
+    public void testReadingNonOrcFile()
+    {
+        assertUpdate("CREATE TABLE textfile (f1 bigint) with( format = 'TEXTFILE')");
```
Comment (mbasmanova) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java]: - capitalize SQL keywords
- f1 -> id, perhaps?

```
assertUpdate("CREATE TABLE textfile (id BIGINT) WITH (format = 'TEXTFILE')");
assertUpdate("INSERT INTO textfile VALUES (1), (2), (3)", 3);
assertQuery("SELECT f1 FROM textfile WHERE f1 = 1", "SELECT 1");
assertQuery("SELECT f1 FROM textfile", "SELECT 1 UNION SELECT 2 UNION SELECT 3 ");
```

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java [hunk 4]
```diff
@@ -654,6 +654,16 @@ public void testPathColumn()
         assertQuerySucceeds(session, "SELECT linenumber, \"$path\" FROM lineitem WHERE length(\"$path\") % 2 = linenumber % 2");
     }
 
+    @Test
+    public void testReadingNonOrcFile()
```
Comment (mbasmanova) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java]: perhaps, `testTextfileTableFormat`

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 5]
```diff
@@ -1876,7 +1877,7 @@ public Void visitVariableReference(VariableReferenceExpression variable, Immutab
                                 hivePartitionResult.getEnforcedConstraint(),
                                 hiveBucketHandle,
                                 hivePartitionResult.getBucketFilter(),
-                                createTableLayoutString(session, handle.getSchemaTableName(), hivePartitionResult.getBucketHandle(), TRUE_CONSTANT, domainPredicate))),
+                                createTableLayoutString(session, handle.getSchemaTableName(), hivePartitionResult.getBucketHandle(), TRUE_CONSTANT, domainPredicate), false)),
```
Comment (mbasmanova) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: false should be on a separate line

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java [hunk 6]
```diff
@@ -66,7 +67,8 @@ public HiveTableLayoutHandle(
             @JsonProperty("partitionColumnPredicate") TupleDomain<ColumnHandle> partitionColumnPredicate,
             @JsonProperty("bucketHandle") Optional<HiveBucketHandle> bucketHandle,
             @JsonProperty("bucketFilter") Optional<HiveBucketFilter> bucketFilter,
-            @JsonProperty("layoutString") String layoutString)
+            @JsonProperty("layoutString") String layoutString,
+            @JsonProperty("pushdownFilterEnabled") boolean pushdownFilterEnabled)
```
Comment (mbasmanova) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java]: pushdownFilterEnabled argument belongs before layoutString; layoutString should be the last argument

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java [hunk 7]
```diff
@@ -94,7 +97,8 @@ public HiveTableLayoutHandle(
             TupleDomain<ColumnHandle> partitionColumnPredicate,
             Optional<HiveBucketHandle> bucketHandle,
             Optional<HiveBucketFilter> bucketFilter,
-            String layoutString)
+            String layoutString,
+            boolean pushdownFilterEnabled)
```
Comment (mbasmanova) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java]: same here; layoutString should be the last argument

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java [hunk 8]
```diff
@@ -654,6 +654,16 @@ public void testPathColumn()
         assertQuerySucceeds(session, "SELECT linenumber, \"$path\" FROM lineitem WHERE length(\"$path\") % 2 = linenumber % 2");
     }
 
+    @Test
+    public void testTextfileTableFormat()
+    {
+        assertUpdate("CREATE TABLE textfile (id BIGINT) WITH (format = 'TEXTFILE')");
+        assertUpdate("INSERT INTO textfile VALUES (1), (2), (3)", 3);
+        assertQuery("SELECT f1 FROM textfile WHERE f1 = 1", "SELECT 1");
```
Comment (mbasmanova) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java]: f1 -> id

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java [hunk 9]
```diff
@@ -654,6 +654,24 @@ public void testPathColumn()
         assertQuerySucceeds(session, "SELECT linenumber, \"$path\" FROM lineitem WHERE length(\"$path\") % 2 = linenumber % 2");
     }
 
+    @Test
+    public void testTextfileAndOrcTableFormatsWithPushdown()
```
Comment (mbasmanova) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java]: @bhhari Now, that you are testing 2 formats, the question is why not test all of them in a loop? See TestHiveIntegrationSmokeTest.

```
    @Test
    public void testCreateEmptyBucketedPartition()
    {
        for (TestingHiveStorageFormat storageFormat : getAllTestingHiveStorageFormat()) {
            testCreateEmptyBucketedPartition(storageFormat.getFormat());
        }
    }
```

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java [hunk 10]
```diff
@@ -600,11 +600,11 @@ public void testPushdownComposition()
         // Tests composing two pushdowns each with a range filter and filter function.
         assertQuery(
                 "WITH data AS (" +
-                "    SELECT l.suppkey, l.linenumber, l.shipmode, MAX(o.orderdate)" +
-                "    FROM lineitem l,  orders o WHERE" +
-                "        o.orderkey = l.orderkey AND linenumber IN (2, 3, 4, 6) AND shipmode LIKE '%AIR%'" +
-                "        GROUP BY l.suppkey, l.linenumber, l.shipmode)" +
-                "SELECT COUNT(*) FROM data WHERE suppkey BETWEEN 10 AND 30 AND shipmode LIKE '%REG%'");
+                        "    SELECT l.suppkey, l.linenumber, l.shipmode, MAX(o.orderdate)" +
```
Comment (mbasmanova) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java]: unrelated change? please, remove it

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java [hunk 11]
```diff
@@ -165,10 +165,12 @@ public MapFlatSelectiveStreamReader(
         }
         else {
             switch (keyOrcTypeKind) {
+                case BOOLEAN:
```
Comment (mbasmanova) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java]: unrelated change? Should it be a separate commit? What is this for?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownIntegrationSmokeTest.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        super(() -> createQueryRunner(ImmutableList.of(ORDERS, CUSTOMER, LINE_ITEM, PART_SUPPLIER), ImmutableMap.of("experimental.pushdown-subfields-enabled", "true"), "sql-standard", ImmutableMap.of("hive.pushdown-filter-enabled", "true"), Optional.empty()),
+                createBucketedSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))),
+                createMaterializeExchangesSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))),
+                HIVE_CATALOG,
+                new HiveTypeTranslator());
+    }
+
+    @Test
+    public void testMaterializedPartitioning()
```
Comment (mbasmanova) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownIntegrationSmokeTest.java]: are these not working? why? Let's put a TODO to get these fixed.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java (hunk 11 above)
Comment (bhhari) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java]: will move to a separate commit before the test

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownIntegrationSmokeTest.java (hunk 12 above)
Comment (bhhari) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownIntegrationSmokeTest.java]: will add TODO with the failure reasons

File: presto-hive/pom.xml [hunk 13]
```diff
@@ -474,6 +477,7 @@
                         <configuration>
                             <includes>
                                 <include>**/TestHivePushdownFilterQueries.java</include>
+                                <include>**/TestHivePushdownIntegrationSmokeTest.java</include>
```
Comment (mbasmanova) [inline presto-hive/pom.xml]: this change should go into `Enable hive integration tests with pushdown` commit
